test/e2e/service.go

```diff
@@ -95,6 +105,253 @@ var _ = ginkgo.Describe("Services", func() {
+						ObjectMeta: metav1.ObjectMeta{Name: fmt.Sprintf(echoServiceNameTemplate, servicePort)},
+						Spec: v1.ServiceSpec{
+							Ports: []v1.ServicePort{
+								{
+									Name:     "tcp-port",
+									NodePort: int32(servicePort),
+									Port:     int32(serverPodPort),
+									Protocol: v1.ProtocolTCP,
+								},
+								{
+									Name:     "udp-port",
+									NodePort: int32(servicePort),
+									Port:     int32(serverPodPort),
+									Protocol: v1.ProtocolUDP,
+								},
+							},
+							Selector: map[string]string{"app": serverPodName},
+							Type:     v1.ServiceTypeNodePort},
+					}
+					_, err := f.ClientSet.CoreV1().Services(f.Namespace.Name).Create(context.TODO(), &svc, metav1.CreateOptions{})
+					return err
+				}, 60*time.Second, 1*time.Second).Should(gomega.Succeed())
+			})
+
+			// Run queries against the service both with a small (10 bytes + overhead for echo service) and
+			// a large (1420 bytes + overhead for echo service) payload.
+			// The payload is transmitted to and echoed from the echo service for both HTTP and UDP tests.
+			ginkgo.When("tests are run towards the agnhost echo service", func() {
+				ginkgo.It("queries to the nodePort service shall work for TCP", func() {
+					for _, size := range []string{"small", "large"} {
+						for _, serviceNodeIP := range serviceNodeInternalIPs {
+							ginkgo.By(fmt.Sprintf("Sending TCP %s payload to service IP %s "+
+								"and expecting to receive the same payload", size, serviceNodeIP))
+							cmd := fmt.Sprintf("curl --max-time 10 -g -q -s http://%s:%d/echo?msg=%s",
+								serviceNodeIP,
+								servicePort,
+								echoPayloads[size],
+							)
+							framework.Logf("Testing TCP %s with command %q", size, cmd)
+							stdout, err := framework.RunHostCmdWithRetries(
+								clientPod.Namespace,
+								clientPod.Name,
+								cmd,
+								framework.Poll,
+								60*time.Second)
+							framework.ExpectNoError(err, fmt.Sprintf("Testing TCP with %s payload failed", size))
+							gomega.Expect(stdout).To(gomega.Equal(echoPayloads[size]), fmt.Sprintf("Testing TCP with %s payload failed", size))
+						}
+					}
+				})
+
+				// We have 2 possible scenarios - hostNetwork endpoints and non-hostNetwork endpoints.
+				// We will only see fragmentation for large (> 1400 Bytes) packets and non-hostNetwork endpoints.
+				//
+				// hostNetwork endpoints:
+				// Packet from ovn-worker2 to GR_ovn-worker where it will hit LB:
+				// # ovn-nbctl lr-lb-list GR_ovn-worker | grep udp | grep 31206
+				// 083a13aa-13e0-45bb-8a1a-175ce7e9fe81    Service_services    udp        172.18.0.3:31206      10.244.2.3:9894
+				// With routes:
+				// # ovn-nbctl lr-route-list GR_ovn-worker  | grep dst-ip
+				//            10.244.0.0/16                100.64.0.1 dst-ip
+				//                0.0.0.0/0                172.18.0.1 dst-ip rtoe-GR_ovn-worker
+				// This means that the packet will enter port rtoe-GR_ovn-worker, be load-balanced and
```

Comment:
nice explanation!

Reply:
was this intentional ? the fact that we do this on rtoj and not rtoe?

Reply:
cc @andreaskaris @trozet